### PR TITLE
Registration Email

### PR DESCRIPTION
**Pull request by hamzaavvan**
Added registration.html file
Updated Users.functions.ts with registrationEmail()

### CODE REVIEW

File: functions/src/modules/Users.functions.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import * as admin from 'firebase-admin';
+
+const db = admin.database();
+
+/**
+ * Sends registration email when a new user is added to /users/ path
+ * 
+ * @function registrationEmail()
+ */
+export const registrationEmail = functions.database
```
Comment (anantakrishna) [inline functions/src/modules/Users.functions.ts]:
```suggestion
export const sendRegistrationEmail = functions.database
```

File: functions/src/modules/Users.functions.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.ref('/users/{user_id}')
+.onCreate(async (snapshot) => {
+  const user = snapshot.val();
+  const coordinatorConfig = functions.config().coordinator;
```
Comment (anantakrishna) [inline functions/src/modules/Users.functions.ts]:
```suggestion
  const coordinator = functions.config().coordinator;
```

File: email/registration.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<div>
+  <p>All Glories to Sri Sri Guru and Gauranga!</p>
+  <p>Hare Krsna. Dandavat pranams dear {{2.your_name}}.</p>
```
Comment (anantakrishna) [inline email/registration.html]: `2.your_name` will not work. You should refer to the data passed into the template through `params`. See [below](https://github.com/PureBhaktiArchive/audioseva/pull/71/files#diff-3af462e09d881812253c21aa19cc1fa4R21).

File: email/registration.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<div>
```
Comment (anantakrishna) [inline email/registration.html]: See the [sample](https://github.com/PureBhaktiArchive/audioseva/blob/release-1.0/email/sqr-allotment.html) that is linked in the card.

- Is there any `div`?
- What is the purpose of this `div`?
- What is the root element there in the sample?

These questions are not to be answered, just adjust your code.

File: email/registration.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  <p>All Glories to Sri Sri Guru and Gauranga!</p>
+  <p>Hare Krsna. Dandavat pranams dear {{params.user.name}}.</p>
+  <br>
```
Comment (anantakrishna) [inline email/registration.html]: It was explicitly said: “paragraphs should be used instead of plain `br`”.

File: email/registration.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  <p>Thank you for choosing to be a part of Srila Gurudeva’s Audio Seva Team!</p>
+  <p>We will soon email you further details about your seva assignment.</p>
+  <br>
+
+  <p>Your Servants,</p>
+  <p>Audio Seva Team.</p>
```
Comment (anantakrishna) [inline email/registration.html]: Why not to take the common parts from the [sample](https://github.com/PureBhaktiArchive/audioseva/blob/release-1.0/email/sqr-allotment.html)?

File: email/registration.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+</div>
+
+<hr>
+
+<div>
+  <p><b>Name:</b> {{params.user.name}}</p>
+  <p><b>Country:</b> {{params.user.location}}</p>
+  <p><b>Email:</b> {{params.user.emailAddress}}</p>
+  <p><b>Contact number:</b> {{params.user.phoneNumber}}{{if(params.user.isAvailableOnWhatsApp; ", available on WhatsApp")}}</p>
+  <p><b>Languages:</b> {% for language in params.user.languages %} {{language}} {{if(!forloop.Last; ", ")}} {% endfor %}</p>
```
Comment (anantakrishna) [inline email/registration.html]: Isn't there a `join` function in the template language? Just wondering.

Comment (hamzaavvan) [reply email/registration.html]: No, but there's a `join` filter.

**Comment by hamzaavvan** [comment]
Ok fixed as mentioned.
And also converted plain `br` to `p` in this sample part
```
<p>
    --<br />
    Your Servants,<br />
    Pure Bhakti Audio Seva Team.
  </p>
```

File: email/registration.html (hunk 7 above)
Comment (anantakrishna) [reply email/registration.html]: Soo have you used it now?

**Comment by anantakrishna** [comment]
> And also converted plain `br` to `p` in this sample part
> 
> ```
> <p>
>     --<br />
>     Your Servants,<br />
>     Pure Bhakti Audio Seva Team.
>   </p>
> ```

In this particular place `br` is used intentionally, as it should be just a line break. So please revert this change.

**Comment by hamzaavvan** [comment]
> There is some mess of commits in this branch. Do you understand what is done wrong?

You pointing to the commits of @ojhasubham?? Well, I tried to create the branch according to the card slug. Have I done something wrong?

image

**Comment by anantakrishna** [comment]
Yes, I mean all the commits listed in this pull request that are not related to this task. The branch name is okay, but what happens on it is not okay.

In this case you have to force-push this branch so that it is correct. You have to deal with GIT nicely in order to proceed.

**Pull request merged**